cylc/flow/rundb.py

```diff
@@ -609,6 +610,11 @@ def select_task_job_run_times(self, callback):
         for row_idx, row in enumerate(self.connect().execute(stmt)):
             callback(row_idx, list(row))
 
+    def select_task_job_platforms(self):
+        """Return the set of platform names from task_jobs table."""
+        stmt = f"SELECT platform_name FROM {self.TABLE_TASK_JOBS}"
+        return set(i[0] for i in self.connect().execute(stmt))
```

Comment:
SQL has a `DISTINCT` keyword that filters out duplicate items which should be a good bit faster for large flows.

https://www.1keydata.com/sql/sqldistinct.html